### PR TITLE
add /connectivity - endpoint to check the network status of a device

### LINKED ISSUE #34: New Endtype Connectivity
**Summary**
===========
In this issue will add a new **Endpoint** which helps to get the current connectivity status. 

**Changes** 
==========
* Adding new Endpoint
    * /connectivity
      * This will give back if the device is currently connected to a network or not (DATA)

### PR DESCRIPTION
**Pull request by maxl2287**
#### What type of PR is this?

Add one of the following kinds:
* enhancement/feature

#### What this PR does / why we need it:
This PR creates a new endpoint "/connectivity" to check the current network status of a device.

#### Which issue(s) this PR fixes:

Fixes #34

### CODE REVIEW

**Comment by maxl2287** [comment]
Hi @eric-murray,

yes I thought so, too.
That's why I updated the version to 0.6.0-wip.

**Comment by bigludo7** [comment]
@maxl2287  Agree with @eric-murray that we must first stabilize the v0.5 with the notification and the proceed this PR in v0.6. 
Hope we will be able to solve quickly the notification.

One question - Why did you removed CONNECTED_DATA_SMS value ? (the device is connected to the network for Data & SMS usage). Sorry if i missed a discussion on this topic but IMHO it makes sense to keep the value when the device can receive Data &SMS. Thanks

**Comment by maxl2287** [comment]
> One question - Why did you removed CONNECTED_DATA_SMS value ? (the device is connected to the network for Data & SMS usage). Sorry if i missed a discussion on this topic but IMHO it makes sense to keep the value when the device can receive Data &SMS. Thanks

I thought that we can remove it, because it's already inside of CONNECTED_DATA. 
Why? - Because I do not know a usecase, where you are connected to data only without be able to receive SMS.
So CONNECTED_DATA_SMS is for me the same like CONNECTED_DATA.

What do you think about this?

**Comment by bigludo7** [comment]
Thanks @maxl2287 for the quick answer. Got it. I will check internally with my network experts but makes sense to me now.

**Comment by akoshunyadi** [comment]
Decision from the DeviceStatus working group call: this PR should be extended by the connectivity events. However they should already be based on CloudEvents. To avoid conflicts this PR should be based on PR #74 .

**Comment by maxl2287** [comment]
@eric-murray & @bigludo7: 

Depending on the MeetingMinutes (https://github.com/camaraproject/DeviceStatus/blob/main/documentation/MeetingMinutes/MoM-2023-09-13.md)
this PR needs to be part of 0.5.0 right?

I would like to wait until PR #75 is done to do the updates for this PR.

File: code/API_definitions/device-status.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ info:
   license:
     name: Apache 2.0
     url: https://www.apache.org/licenses/LICENSE-2.0.html
-  version: 0.5.0-wip
+  version: 0.6.0-wip
```
Comment (sachinvodafone) [inline code/API_definitions/device-status.yaml:11]: Under description , we can mention "connectivity" too like "This API allows to query device roaming status, **connectivity** and subscribe to event related to device roaming status."

**Comment by maxl2287** [comment]
@sachinvodafone @eric-murray @bigludo7 
I have updated this PR based on CloudEvents.

Please review. :)

**Review by bigludo7 — COMMENTED** [review]
Hello @maxl2287 
Thanks for the proposal.
Few cosmetic change and one code change proposal: The eventType org.camaraproject.device-status.v0.subscription-ends has not to be subscribed (see API description). This notification does not require dedicated subscription. But, the eventType could be send in the notification so I've introduced 2 definition of eventType: one with and one without subscription-ends. 

Hope it makes sense for you

**Comment by bigludo7** [comment]
@maxl2287 This is the same comment for https://github.com/camaraproject/DeviceLocation/issues/113 ;)

**Comment by bigludo7** [comment]
> ```yaml
> SubscriptionDetail
> ```

Hello @maxl2287 
Yes. Sorry for the proposal - we can simplify.

My only request is to have a distinct array for the `type` attribute in `SubscriptionDetail` from the array in `CloudEvent`.

**Comment by maxl2287** [comment]
@bigludo7 sorry I have inserted the wrong code as a proposal 😄 .

I mean: 
Is this here a good way to reduce redundancy:

```yaml
    SubscriptionEventType:
      allOf:
        - $ref: "#/components/schemas/SubscriptionCreationEventType"
        - type: string
          description: |
            subscription-ends: Event triggered when the subscription ends.
          enum:
            - org.camaraproject.device-status.v0.subscription-ends
```

**Comment by bigludo7** [comment]
> @bigludo7 sorry I have inserted the wrong code as a proposal 😄 .
> 
> I mean: Is this here a good way to reduce redundancy:
> 
> ```yaml
>     SubscriptionEventType:
>       allOf:
>         - $ref: "#/components/schemas/SubscriptionCreationEventType"
>         - type: string
>           description: |
>             subscription-ends: Event triggered when the subscription ends.
>           enum:
>             - org.camaraproject.device-status.v0.subscription-ends
> ```

Yes this is excellent. I should confess that I did not know that it was possible to extend an enum this way. Thanks !!

**Review by bigludo7 — DISMISSED** [review]
LGTM
Thanks @maxl2287

**Review by sachinvodafone — DISMISSED** [review]

**Comment by maxl2287** [comment]
> > @bigludo7 sorry I have inserted the wrong code as a proposal 😄 .
> 
> > 
> 
> > I mean: Is this here a good way to reduce redundancy:
> 
> > 
> 
> > ```yaml
> 
> >     SubscriptionEventType:
> 
> >       allOf:
> 
> >         - $ref: "#/components/schemas/SubscriptionCreationEventType"
> 
> >         - type: string
> 
> >           description: |
> 
> >             subscription-ends: Event triggered when the subscription ends.
> 
> >           enum:
> 
> >             - org.camaraproject.device-status.v0.subscription-ends
> 
> > ```
> 
> 
> 
> Yes this is excellent. I should confess that I did not know that it was possible to extend an enum this way. Thanks !!

I don't know. I thought so, but in DeviceStatus @jlurien denied that possibility.

So I guess I need to revert it.

**Comment by maxl2287** [comment]
@bigludo7 I have reverted my commit and deleted an unused component.
Now I think its good to be reviewed.

I was not aware that an "extend" is not possible for enums yet and it's only possible by a bit of redundancy.

**Review by bigludo7 — APPROVED** [review]
LGTM. Thanks @maxl2287

**Review by sachinvodafone — APPROVED** [review]

**Pull request merged**